### ISSUE TITLE
`cargo clippy` invocation fails to properly parse arguments: `error: Unrecognized option: 'all-packages'`

### ISSUE THREAD

**Issue by willbuckner** [rust-lang/rust-clippy#9079]
### Summary

After updating to Rust 1.62 stable (Clippy 0.1.62), `cargo clippy` fails to execute under multiple scenarios:

```
❱ cargo clippy
error: failed to run `rustc` to learn about target-specific information

Caused by:
  process didn't exit successfully: `/Users/will/.rustup/toolchains/stable-aarch64-apple-darwin/bin/clippy-driver rustc - --crate-name ___ --print=file-names --crate-type bin --crate-type rlib --crate-type dylib --crate-type cdylib --crate-type staticlib --crate-type proc-macro --print=sysroot --print=cfg` (exit status: 1)
  --- stderr
  error: Unrecognized option: 'all-packages'
❱ 
```

```
❱ command cargo +stable clippy
error: failed to run `rustc` to learn about target-specific information

Caused by:
  process didn't exit successfully: `/Users/will/.rustup/toolchains/stable-aarch64-apple-darwin/bin/clippy-driver rustc - --crate-name ___ --print=file-names --crate-type bin --crate-type rlib --crate-type dylib --crate-type cdylib --crate-type staticlib --crate-type proc-macro --print=sysroot --print=cfg` (exit status: 1)
  --- stderr
  error: Unrecognized option: 'all-packages'
❱ 
```

```
❱ command cargo clippy --all-targets --all-features -- -D warnings
error: failed to run `rustc` to learn about target-specific information

Caused by:
  process didn't exit successfully: `/Users/will/.rustup/toolchains/stable-aarch64-apple-darwin/bin/clippy-driver rustc - --crate-name ___ --print=file-names --crate-type bin --crate-type rlib --crate-type dylib --crate-type cdylib --crate-type staticlib --crate-type proc-macro --print=sysroot --print=cfg` (exit status: 1)
  --- stderr
  error: Unrecognized option: 'all-packages'

❱
```

The project was originally compiled with Rust 1.61 stable toolchain, and all targets had previously been built with 1.61. After the `rustup update`, I initially ran a simple `cargo clippy` which recompiled and ran clippy successfully.

However, attempting to use `cargo clippy --all-packages` the first time resulted in `error: Unrecognized option: 'all-packages'`. All subsequent clippy commands resulted in the same `error: Unrecognized option: 'all-packages'`, though the errant option was never passed to these invocations. A `cargo clean` fixed the issue, and I haven't been able to reproduce again.

@jyn514 was able to reproduce this issue on x86_64 as well:

```
> command cargo +stable clippy 
error: failed to run `rustc` to learn about target-specific information

Caused by:
  process didn't exit successfully: `/home/jnelson/.local/lib/rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/clippy-driver rustc - --crate-name ___ --print=file-names --crate-type bin --crate-type rlib --crate-type dylib --crate-type cdylib --crate-type staticlib --crate-type proc-macro --print=sysroot --print=cfg` (exit status: 1)
  --- stderr
  error: Unrecognized option: 'all-targets'
```

Thanks! Let me know if there's any additional information I can provide, or tests I can run.

### Version

```text
rustc 1.62.0 (a8314ef7d 2022-06-27)
binary: rustc
commit-hash: a8314ef7d0ec7b75c336af2c9857bfaf43002bfc
commit-date: 2022-06-27
host: aarch64-apple-darwin
release: 1.62.0
LLVM version: 14.0.5

clippy 0.1.62 (a8314ef7 2022-06-27)
```

**Comment by vnermolaev** [comment]
Same here
```shell
cargo clippy                             
error: failed to run `rustc` to learn about target-specific information

Caused by:
  process didn't exit successfully: `/Users/vic/.rustup/toolchains/stable-x86_64-apple-darwin/bin/clippy-driver rustc - --crate-name ___ --print=file-names --crate-type bin --crate-type rlib --crate-type dylib --crate-type cdylib --crate-type staticlib --crate-type proc-macro --print=sysroot --print=cfg` (exit status: 1)
  --- stderr
  error: Unrecognized option: 'all-features'
```

```                     
rustc 1.64.0 (a55dd71d5 2022-09-19)
clippy 0.1.64 (a55dd71d 2022-09-19)
host: x86_64-apple-darwin
```

**Comment by Qix-** [comment]
Running into this myself. Running clippy using `--all --examples` causes issues in one directory (`Unrecognized option: examples`) but not another. It was working fine in the faulty directory until I edited an example, and then out of the blue stopped working. No toolchain changes, nada. Nothing in Cargo.toml/.lock indicates that a different version of Clippy would be pulled.

Huge blocker :( No idea how to even debug this.

**Comment by Haradric** [comment]
I get the same issue after running:

```bash
cargo clippy -- --all-targets --all-features --tests -D warnings -W clippy::pedantic -W clippy::nursery -W rust-2018-idioms
```

Instead of:

```bash
cargo clippy --all-targets --all-features --tests -- -D warnings -W clippy::pedantic -W clippy::nursery -W rust-2018-idioms
```

After this, any clippy invocation came with this error. Deleting the `target` directory resolved the issue.
